src/js/App/Sidenav/Navigation/DynamicNav.js

```diff
@@ -0,0 +1,44 @@
+import React, { useState, useEffect } from 'react';
+import { useModule } from '@scalprum/react-core';
+import PropTypes from 'prop-types';
+import { Skeleton } from '@patternfly/react-core';
+import ChromeNavItemFactory from './ChromeNavItemFactory';
+import { useHistory } from 'react-router';
+import { useDispatch, useSelector } from 'react-redux';
+import { loadLeftNavSegment } from '../../../redux/actions';
+
+const DynamicNav = ({ dynamicNav, ...rest }) => {
+  const [nav, setNav] = useState();
+  const {
+    location: { pathname },
+  } = useHistory();
+  const dispatch = useDispatch();
+  const [appName] = dynamicNav.split('/');
+  const currentNamespace = pathname.split('/')[1];
+  const schema = useSelector(({ chrome: { navigation } }) => navigation[currentNamespace]);
+  const { default: navigation } = useModule(appName, './Navigation', {});
+  useEffect(() => {
+    if (navigation) {
+      if (typeof navigation === 'function') {
+        Promise.resolve(navigation({ schema, dynamicNav, currentNamespace })).then((data) => {
+          schema.navItems = schema?.navItems?.map((item, key) => ({
+            ...item,
+            ...(key === schema.navItems.findIndex((nav) => nav.dynamicNav === dynamicNav) && data),
+          }));
+          dispatch(loadLeftNavSegment(schema, currentNamespace, window.location.pathname));
```

Comment:
Is there a reason to use window location rather than router location? The only difference between those two should only ever be the `/beta` prefix.